src/TextInputWidget.cc

```diff
@@ -233,11 +228,40 @@ FilteredTextEdit::textChanged()
 }
 
 void
-FilteredTextEdit::receiveImage(const QByteArray img, const QString &img_name)
+FilteredTextEdit::uploadData(const QByteArray data, const QString &media, const QString &filename)
 {
         QSharedPointer<QBuffer> buffer{new QBuffer{this}};
-        buffer->setData(img);
-        emit image(buffer, img_name);
+        buffer->setData(data);
+
+        if (media == "image")
+                emit image(buffer, filename);
+        else if (media == "audio")
+                emit audio(buffer, filename);
+        else if (media == "video")
+                emit video(buffer, filename);
+        else
+                emit file(buffer, filename);
+}
+
+void
+FilteredTextEdit::showPreview(const QMimeData *source,
+                              const QStringList &formats,
+                              const QString &media,
+                              const QString &type)
+{
+        const auto idx = formats.indexOf(
+          QRegularExpression{media + "/.+", QRegularExpression::CaseInsensitiveOption});
```

Comment:
Do we really need to search again, when in practice `formats` always contains nothing but matches and is nonempty?

Reply:
I presume this comment relies on the previous comment, but I'll reply anyways.

This function's purpose is to be a helper to search the formats list, and if that fails, have the correct fall-back, and retrieve the data (as QByteArray) and send it on over to PreviewUploadOverlay. It is purely for not having to rewrite this logic in every if-statement.

Reply:
The formats list that you pass in seems to always have already been filtered such that this search is meaningless, and a fallback will never occur.